lib/Conversion/TritonGPUToLLVM/ViewOpToLLVM.cpp

```diff
@@ -298,23 +298,21 @@ struct TransOpConversion : public ConvertOpToLLVMPattern<TransOp> {
   LogicalResult
   matchAndRewrite(TransOp op, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
+    // TODO: With linear layouts, this should just return adaptor.getSrc()
+    // We should check whether this is also the case for the legacy path
```

Comment:
I think you are right, I can't think of a case where it wouldn't, the unpackLLElements and packLLElements will assume the same number of elements so I think any other case would have been broken